modules/tinymce/src/core/main/ts/mode/Disabled.ts

```diff
@@ -0,0 +1,140 @@
+import { Arr, Optional, Strings } from '@ephox/katamari';
+import { Attribute, Compare, ContentEditable, SelectorFilter, SelectorFind, SugarElement } from '@ephox/sugar';
+
+import Editor from '../api/Editor';
+import VK from '../api/util/VK';
+import * as ModeUtils from '../util/ModeUtils';
+
+const isDisabled = (editor: Editor): boolean => editor.disabled;
+
+const internalContentEditableAttr = 'data-mce-contenteditable';
+
+const switchOffContentEditableTrue = (elm: SugarElement<Node>) => {
+  Arr.each(SelectorFilter.descendants<HTMLElement>(elm, '*[contenteditable="true"]'), (elm) => {
+    Attribute.set(elm, internalContentEditableAttr, 'true');
+    ContentEditable.set(elm, false);
+  });
+};
+
+const switchOnContentEditableTrue = (elm: SugarElement<Node>) => {
+  Arr.each(SelectorFilter.descendants<HTMLElement>(elm, `*[${internalContentEditableAttr}="true"]`), (elm) => {
+    Attribute.remove(elm, internalContentEditableAttr);
+    ContentEditable.set(elm, true);
+  });
+};
+
+const toggleDisabled = (editor: Editor, state: boolean): void => {
+  const body = SugarElement.fromDom(editor.getBody());
+
+  if (state) {
+    editor.disabled = true;
+    ModeUtils.disableEditor(editor);
+    ContentEditable.set(body, false);
```

Comment:
Do we have to undo this in the `else` block or that done somewhere else?

Reply:
I've moved that to the `ModeUtils.enableEditor` since some of the functionalities are shared in the disabled mode and readonly mode.